include/fmt/core.h

```diff
@@ -3071,7 +3071,7 @@ FMT_API auto vformat(string_view fmt, format_args args) -> std::string;
   \endrst
 */
 template <typename... T>
-FMT_INLINE auto format(format_string<T...> fmt, T&&... args) -> std::string {
+[[nodiscard]] FMT_INLINE auto format(format_string<T...> fmt, T&&... args) -> std::string {
```

Comment:
Probably needs a macro for ancient compilers which can't ignore unknown attributes.